content/doc/book/pipeline/syntax.adoc

```diff
@@ -0,0 +1,851 @@
+
+
+[[parameters-example]]
+===== Example
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent any
+    parameters {
+        string(name: 'PERSON', defaultValue: 'Mr Jenkins', description: 'Who should I say hello to?'
+    }
+    stages {
+        stage('Example') {
+            steps {
+                echo "Hello ${params.PERSON}"
+            }
+        }
+    }
+}
+----
+
+[NOTE]
+====
+A comprehensive list of available parameters is pending the completion of
+link:https://issues.jenkins-ci.org/browse/INFRA-1053[INFRA-1503].
+====
+
+==== triggers
+
+The `triggers` directive defines the automated ways in which the Pipeline
+should be re-triggered. For Pipelines which are integrated with a source such
+as GitHub or BitBucket, `triggers` may not be necessary as webhooks-based
+integration will likely already be present. Currently the only two available
+triggers are `cron` and `pollScm`.
+
+[cols="^10h,>90a",role=syntax]
+|===
+| Required
+| No
+
+| Parameters
+| _None_
+
+| Allowed
+| Only once, inside the `pipeline` block.
+|===
+
+
+cron:: Accepts a cron-style string to define a regular interval at which the
+Pipeline should be re-triggered, for example: `triggers { cron('@daily') }`
+pollScm:: Accepts a cron-style string to define a regular interval at which
+Jenkins should check for new source changes. If new changes exist, the Pipeline
+will be re-triggered. For example: `triggers { pollScm('@daily') }`
+
+[NOTE]
+====
+The `pollScm` trigger is only available in Jenkins 2.22 or later.
+====
+
+[[triggers-example]]
+===== Example
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent any
+    triggers {
+        cron('@daily')
+    }
+    stages {
+        stage('Example') {
+            steps {
+                echo 'Hello World'
+            }
+        }
+    }
+}
+----
+
+==== stage
+
+The `stage` directive goes in the `stages` section and should contain one or
+more steps, an optional `agent` directive, or other stage specific directives.
+Practically speaking, all of the real work done by a Pipeline will be wrapped
+in one or more `stage` directives.
+
+[cols="^10h,>90a",role=syntax]
+|===
+| Required
+| At least one
+
+| Parameters
+| A single String, the name of stage.
+
+| Allowed
+| Inside the `stages` section.
+|===
+
+[[stage-example]]
+===== Example
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent any
+    stages {
+        stage('Example') {
+            steps {
+                echo 'Hello World'
+            }
+        }
+    }
+}
+----
+
+==== tools
+////
+XXX: This is intentionally light until
+https://issues.jenkins-ci.org/browse/WEBSITE-193
+////
+
+A section defining tools to auto-install and put on the `PATH`. This is ignored
+if `agent none` is specified.
+
+[cols="^10h,>90a",role=syntax]
+|===
+| Required
+| No
+
+| Parameters
+| _None_
+
+| Allowed
+| Only once, inside the `pipeline` block.
```

Comment:
Shouldn't this be allowed wherever an `agent` is allowed?  If I need a tool during one stage, with a particular agent, does this mean that I'm forced to have the tool auto-installed on all agents used during the Pipeline?